game/sony/PS/PSP/psp_save_metadata.ksy

```diff
@@ -0,0 +1,102 @@
+meta:
+  id: psp_save_metadata
+  endian: le
+  encoding: utf-8
+  license: Unlicense
+  file-extension: SFO
+doc: |
+  PSP save metadata (PARAM.SFO)
+seq:
+  - id: header
+    type: header
+  - id: count
+    type: u4
+  - id: descriptors
+    type: descriptor
+    repeat: expr
+    repeat-expr: count
+  - id: names
+    type: strz
+    repeat: expr
+    repeat-expr: count
+  - id: unkn0
+    type: u1
+  - id: sections
+    type: section(descriptors[_index].size, names[_index])
+    repeat: expr
+    repeat-expr: count
+types:
+  section:
+    params:
+      - id: size
+        type: u4
+      - id: name
+        type: str
+    seq:
+      - id: section
+        size: size
+        type:
+          switch-on: name
+          cases:
+            "'CATEGORY'": category
+            "'PARENTAL_LEVEL'": u4 # I guess it is min allowed age
+            "'SAVEDATA_DETAIL'": strz # the string displayed in save description
+            "'SAVEDATA_DIRECTORY'": strz # the name of the directory where a save is saved
+            "'SAVEDATA_FILE_LIST'": savedata_file_list
+            "'SAVEDATA_PARAMS'": savedata_params # don't know what is it
+            "'SAVEDATA_TITLE'": strz # arbitrary string, usually contains a name of a player or a anme of last played level
+            "'TITLE'": strz # contains a name of a game
```

Comment:
Mixing strings, integers and user types would actually break on languages like C++, which do not have an equivalent of `Object` or `Any` type. I'm not sure if we care, though.

Reply:
We care, but we don't. I mean this is compiler bug and compiler should detect and fix this situation automatically. For example it can for c++ generate a special object keeping the type and a pointer and having the conversoin operators (and conversion move operators!) redefined.